docs/design/2022-12-06-support-late-materialization

```diff
@@ -0,0 +1,142 @@
+# Proposal: Support late materialization
+
+* Author: @Lloyd-Pottiger
+* Tracking issue: 
+
+## Background
+
+TiFlash is the key component that makes TiDB essentially an Hybrid Transactional/Analytical Processing (HTAP) database. 
+
+Operating on positions opens a number of different query processing strategies centered around so-called "materialization". Roughly speaking, materialization is the process of switching positions back into values. The literature presents two main strategies: early and late materialization. The basic idea of late materialization is to operate on positions and defer tuple reconstruction for as long as possible. Refer to [A Comprehensive Study of Late Materialization Strategies for a Disk-Based Column-Store](https://ceur-ws.org/Vol-3130/paper3.pdf). 
+
+There are two kinds of late materialization:
+1. Late materialization in selection
+
+    ![Fig.1](./imgs/late-materialization-in-selection.png)
+
+2. Late materialization in join （#Fig.2）
+
+    ![Fig.2](./imgs/late-materialization-in-join.png)
+
+**Only the late materialization in selection will be considered this time.**
+
+Late materialization can help improve performance of some AP queries. Like `SELECT * FROM t WHERE col_str LIKE '%PingCAP%'` The normal execution processes are as follows:
+- Read all the data pack of all the columns on table t
+- Run the filter `col_str LIKE '%PingCAP%'`
+- Return the result
+
+With late materialization, the execution processes can be:
+- Read all the data pack of col_str
+- Run the filter `col_str LIKE '%PingCAP%'`
+- Read the needed data pack of the rest columns on table t
+- Return the result
+
+And for more complex queries, like `SELECT * FROM t WHERE col_str1 LIKE '%PingCAP%' AND REGEXP_LIKE(col_str2, 'new\\*.\\*line')`, late materialization, the execution processes can be:
+- Read all the data pack of col_str
+- Run the filter `col_str1 LIKE '%PingCAP%'`
+- Read the needed data pack of rest columns on table t
+- Run the filter `REGEXP_LIKE(col_str2, 'new\\*.\\*line')`
```

Comment:
After filtering col_str1, why we read all the rest columns instead of only col_str2 column?